xdsl/ir/core.py

```diff
@@ -423,7 +425,18 @@ def print_parameter(self, printer: Printer):
 class ParametrizedAttribute(Attribute):
     """An attribute parametrized by other attributes."""
 
-    def __init__(self, parameters: Sequence[Attribute] = ()):
+    def __init__(self, *parameters: Attribute):
```

Comment:
This will need to be typed `Attribute | Sequence[Attribute]` right? likely with some overloads above?

Reply:
Nah I think we want to discourage this completely, I just don't want to break the user's code, but I'm happy to break their type checking.

Reply:
Perhaps it's fine for the type to not include the `Sequence` as the deprecation warning should still fire if you run `make tests` or equivalent

Reply:
I think so